### PR TITLE
Update the supported configurations.

### PR DESCRIPTION
**Pull request by david-german-tri**
- Be explicit about which Java versions are supported.
- Be explicit about which MATLAB versions are supported.
- Formalize support for MATLAB on Windows and OS X.
- Formalize support for Ninja.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/2534)

### CODE REVIEW

File: drake/doc/developers.rst [hunk 1]
```diff
@@ -52,9 +52,12 @@ The following table shows the configurations and platforms that Drake
...
 
-On Ubuntu and OS X, only the "Unix Makefiles" CMake generator is supported.
-On Windows, only the "Visual Studio 14 2015" and "Visual Studio 14 2015 Win64"
-CMake generators are supported.
+On Ubuntu and OS X, the "Unix Makefiles" and "Ninja" CMake generators are
+supported. On Windows, the "Visual Studio 14 2015",
+"Visual Studio 14 2015 Win64", and "Ninja" CMake generators are supported.
+
+The supported version of MATLAB is R2015b. The supported version of the
+Java Development Kit is 7 on Windows and Ubuntu, and 8 on OS X.
```
Comment (jamiesnape) [inline drake/doc/developers.rst]: Personally, I would just say JDK 7 for all platforms. I can tweak the OS X configuration on CI easily enough. Much of me would like to say that the minimum version is also 7 for all platforms, which realistically would mean R2013b would be minimum MATLAB version.

**Comment by jamiesnape** [comment]
:lgtm:

<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary>
> #### Update the supported configurations\.
> 
> - Be explicit about which Java versions are supported.
> 
> - Be explicit about which MATLAB versions are supported.
> 
> - Formalize support for MATLAB on Windows and OS X.
> 
> - Formalize support for Ninja.
> 
> 
> 
</details>


---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2534#-:-KJr2ARCe-UHKVxu38WO:1417088721)*

**Comment by david-german-tri** [comment]
Done.

<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary>
> :lgtm:
</details>


---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2534#-:-KJr8EtxQe5HIsbRUA2J:-499146009)*

**Comment by jamiesnape** [comment]
:lgtm:

<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary>
> Done.
</details>


---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2534#-:-KJr8g6htfM79_irBAt9:1417088721)*

**Comment by RussTedrake** [comment]
Lgtm

**Comment by david-german-tri** [comment]
Given @russtedrake's emoticon-less approval, I'm going to self-approve this so reviewable will permit the merge. :lgtm:

<details><summary><i>Previously, RussTedrake (Russ Tedrake) wrote…</i></summary>
> Lgtm
</details>


---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/2534#-:-KK8xm5Af1-uQVc1ecs7:777534452)*

**Pull request merged**